ask-sdk-dynamodb-persistence-adapter/ask_sdk_dynamodb/partition_keygen.py

```diff
@@ -59,3 +59,32 @@ def device_id_partition_keygen(request_envelope):
     except AttributeError:
         raise PersistenceException("Couldn't retrieve device id from "
                                    "request envelope, for partition key use")
+
+
+def person_id_partition_keygen(request_envelope):
+    # type: (RequestEnvelope) -> str
+    """Retrieve person id from request envelope, to use as object key.
+
+    This method retrieves the `person id` specific to a voice profile from
+    the request envelope if it exists, to be used as an object key. If it
+    doesn't exist, then the `user id` is returned instead.
+
+    :param request_envelope: Request Envelope passed during skill
+        invocation
+    :type request_envelope: ask_sdk_model.RequestEnvelope
+    :return: Person Id retrieved from request envelope if exists, else
+        fall back on User Id
+    :rtype: str
+    :raises: :py:class:`ask_sdk_core.exceptions.PersistenceException`
+    """
+    try:
+        person_id = request_envelope.context.system.person.person_id
+    except AttributeError:
```

Comment:
Generally a bad practice to use [exceptions for flow control](https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl) but it appears in Python this is an _acceptable_ practice. :confounded:

Reply:
Calling a function to attempt to get the attribute (else fallback to userId) might solve the flow control issue.